src/App.tsx

```diff
@@ -1,19 +1,122 @@
+/* eslint-disable prefer-destructuring */
+/* eslint-disable react/prefer-stateless-function */
 import React from 'react';
 import './App.css';
-
 import users from './api/users';
+import todos from './api/todos';
+import { TodoList } from './Components/TodoList';
+import { Todo } from './Types/Todo';
+import { User } from './Types/User';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  tasks: Todo[],
+  taskName: string,
+  userName: string,
 };
 
+const completedList: Todo[] = todos.map((todo) => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || null,
+}));
+
+class App extends React.Component<{}, State> {
+  state = {
+    tasks: completedList,
+    taskName: '',
+    userName: 'Please choose a user',
+  };
+
+  createTask() {
+    if (this.validate()) {
```

Comment:
лучше делать проверку, и если данные не валидны - возвращать ошибку и выходить из функции. Тогда можно не оборачивать всю основную логику в if
```suggestion
    if (!this.validate()) {
      // error
      
      return;
    }
```